image.go

```diff
@@ -130,7 +137,14 @@ func (i *image) IsUnpacked(ctx context.Context, snapshotterName string) (bool, e
 	return false, nil
 }
 
-func (i *image) Unpack(ctx context.Context, snapshotterName string) error {
+func (i *image) Unpack(ctx context.Context, snapshotterName string, opts ...ImageUnpackOpt) error {
```

Comment:
There should be an ImageUnpackOpt with `[]diff.ApplyOpts` so that these are passed down without knowing the type in the image.go code

Reply:
The following is from `/image.go`'s `Unpack()`
```
[...]
        layers, err := i.getLayers(ctx, i.platform)
[...]
        for _, layer := range layers {
                iopts.LayerToolDecryptData.Descriptor = layer.Blob
                any, err := typeurl.MarshalAny(&iopts.LayerToolDecryptData)
                if err != nil {
                        return errors.Wrapf(err, "failed to typeurl.MarshalAny(LayerToolDecryptData)")
                }

                payloads := make(map[string]*types.Any)
                payloads["io.containerd.layertool.tar"] = any
                payloads["io.containerd.layertool.tar.gzip"] = any
                applyOpts := []diff.ApplyOpt{diff.WithPayloads(payloads)}

                unpacked, err = rootfs.ApplyLayerWithOpts(ctx, layer, chain, sn, a, nil, applyOpts)
                if err != nil {
                        return err
                }
```

The issue here is that layer.Blob only becomes available here. It holds the Annotations that we need to pass to the layertool. How do I pull this all out of this function into the caller ? Should I pass a callback that this function invokes?

Reply:
The callback could then return `[]diff.ApplyOpts`.